packages/patrol_cli/lib/src/features/drive/test_runner.dart

```diff
@@ -175,11 +184,13 @@ class TestRunner extends Disposable {
             continue;
           }
 
-          try {
-            await builder(target, device);
-          } catch (_) {
-            targetRuns.add(TargetRunStatus.failedToBuild);
-            continue;
+          if (_useApplicationBinary == null) {
+            try {
+              await builder(target, device);
+            } catch (_) {
+              targetRuns.add(TargetRunStatus.failedToBuild);
+              continue;
+            }
```

Comment:
make an early return ("_useApplicationBinary != null; then return")

Reply:
I'm not sure we're on the same page here. 🤔
We checked whether _useApplicationBinary (from parseInput() then config object).
If it was specified, then we skip the building part only.
The rest should continue to execute normally.